bitcoin-magazine/bitcoin-explained/explaining-bitcoin-addresses.md

```diff
@@ -0,0 +1,649 @@
+
+Sorry, public script hash. Well, not even public, just pay to script hash.
+
+Aaron Van Wirdum: 00:16:14
+
+We're getting there.
+
+Sjors Provoost: 00:16:15
+
+We're getting there.
+
+Aaron Van Wirdum: 00:16:16
+
+Eventually, pay to script hash.
+
+Sjors Provoost: 00:16:17
+
+Yes, and it says basically anybody who has the script belonging to this hash and who can satisfy the script. So just knowing the script is not enough, you actually have to do whatever the script says you should do.
+
+Aaron Van Wirdum: 00:16:30
+
+Yeah, so the first version we just described was payToPublicKeyHash, which required people to offer a valid signature corresponding to the public key. And now we're talking about payToScriptHash, which means someone needs to present the script and be able to solve the script. So why do these start with a three?
+
+Sjors Provoost: 00:16:51
+
+There's just a convention. So as we said, there is basically a version number that everything that you communicate through Base50H starts with a version number. And if it starts with a 1 then you know it's paid public key hash if it starts with a 3 you know it's paid a script hash if it starts with a 5 you know it's a private key.
+
+Sjors Provoost: 00:17:09
+
+So it's just a convention and once again meaning on the blockchain itself
+
+Aaron Van Wirdum: 00:17:13
+
+once again all this is is a version number plus this hash represented in base 58. Is that all it is? Yeah. This is all so much simpler than I once thought, Sjors.
+
+Sjors Provoost: 00:17:26
+
+No, it's really simple. And the only mystery that has been solved today, I guess, is, well, what if you only use the public key but that wasn't done using this system so there is no initial letter that would represent trying to do that
+
+Aaron Van Wirdum: 00:17:40
+
+yeah that was never represented in base68
+
+Sjors Provoost: 00:17:42
+
+otherwise probably that would have been version 0 and then all normal addresses might have started with the two who knows okay
+
+Aaron Van Wirdum: 00:17:48
+
+I think for anyone who already knew this which is probably a good chunk of people this is a very boring episode so far but I think it's gonna get better because oh my god because Sjors we now have a new type of address since a year or two, which starts with BC1.
+
+Sjors Provoost: 00:18:09
+
+BC1Q even, usually.
+
+Aaron Van Wirdum: 00:18:11
+
+Yeah, usually, but not always. And we're getting into that, I think. Yep. So what is this all about?
+
+Sjors Provoost: 00:18:16
+
+Well, that is BEG32 or BECH32 or however you want to pronounce it. And it's been used since SegWit, basically. And again, it is something that doesn't exist on the blockchain, so it's just a convention that wallets can use. This is a, as the name suggests, a base 32 system. Which means you have almost all the letters, and almost all the numbers, minus some ambiguous characters that you don't want to have, because they look too much like numbers or letters.
```

Comment:
```suggestion
Well, that is BECH32 or however you want to pronounce it. And it's been used since SegWit, basically. And again, it is something that doesn't exist on the blockchain, so it's just a convention that wallets can use. This is a, as the name suggests, a base 32 system. Which means you have almost all the letters, and almost all the numbers, minus some ambiguous characters that you don't want to have, because they look too much like numbers or letters.
```